Update to Quarkus 2.14.1.Final

### Description

The Quarkus team released 2.14.0.Final yesterday with some bug fixes, we should be able to update.

Related issues: #14925

i got issues upgrading 2.14.0
tried: https://quarkus.io/guides/security-keycloak-authorization
but changing back to 2.13.4 worked again

Caused by: org.keycloak.authorization.client.util.HttpResponseException: Unexpected response from server: 400 / Bad Request / Response from server: {"error":"unauthorized_client","error_description":"Unexpected error when authenticating client"}

Just following the discussion

--- a/pom.xml
+++ b/pom.xml
@@ -41,7 +41,7 @@
         <jboss.snapshots.repo.id>jboss-snapshots-repository</jboss.snapshots.repo.id>
         <jboss.snapshots.repo.url>https://s01.oss.sonatype.org/content/repositories/snapshots/</jboss.snapshots.repo.url>
 
-        <quarkus.version>2.13.3.Final</quarkus.version>
+        <quarkus.version>2.14.1.Final</quarkus.version>
 
         <project.build-time>${timestamp}</project.build-time>
 
@@ -83,7 +83,6 @@
         <hibernate.c3p0.version>5.6.10.Final</hibernate.c3p0.version>
         <infinispan.version>14.0.2.Final</infinispan.version>
         <infinispan.protostream.processor.version>4.5.0.Final</infinispan.protostream.processor.version>
-        <javax.annotation-api.version>1.3.2</javax.annotation-api.version>
         <jackson.version>2.13.4</jackson.version>
         <jackson.databind.version>2.13.4.2</jackson.databind.version>
         <jackson.annotations.version>${jackson.version}</jackson.annotations.version>
@@ -165,7 +164,7 @@
         <!-- KEYCLOAK-17585 Prevent microprofile-metrics-api upgrades from version "2.3" due to:
              https://issues.redhat.com/browse/KEYCLOAK-17585?focusedCommentId=16002705&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16002705
         -->
-        <microprofile-metrics-api.version>3.0.1</microprofile-metrics-api.version>
+        <microprofile-metrics-api.version>2.3</microprofile-metrics-api.version>
         <testcontainers.version>1.16.3</testcontainers.version>
 
         <!-- Maven Plugins -->

--- a/pom.xml
+++ b/pom.xml
@@ -83,6 +83,7 @@
         <hibernate.c3p0.version>5.6.10.Final</hibernate.c3p0.version>
         <infinispan.version>14.0.2.Final</infinispan.version>
         <infinispan.protostream.processor.version>4.5.0.Final</infinispan.protostream.processor.version>
+        <javax.annotation-api.version>1.3.2</javax.annotation-api.version>
         <jackson.version>2.13.4</jackson.version>
         <jackson.databind.version>2.13.4.2</jackson.databind.version>
         <jackson.annotations.version>${jackson.version}</jackson.annotations.version>
@@ -95,7 +96,7 @@
         <jboss.spec.javax.xml.bind.jboss-jaxb-api_2.3_spec.version>2.0.1.Final</jboss.spec.javax.xml.bind.jboss-jaxb-api_2.3_spec.version>
         <jboss.spec.javax.servlet.jsp.jboss-jsp-api_2.3_spec.version>2.0.0.Final</jboss.spec.javax.servlet.jsp.jboss-jsp-api_2.3_spec.version>
         <log4j.version>1.2.17</log4j.version>
-        <resteasy.version>4.7.4.Final</resteasy.version>
+        <resteasy.version>4.7.7.Final</resteasy.version>
         <resteasy.undertow.version>${resteasy.version}</resteasy.undertow.version>
         <owasp.html.sanitizer.version>20211018.2</owasp.html.sanitizer.version>
         <slf4j-api.version>1.7.30</slf4j-api.version>
@@ -164,7 +165,7 @@
         <!-- KEYCLOAK-17585 Prevent microprofile-metrics-api upgrades from version "2.3" due to:
              https://issues.redhat.com/browse/KEYCLOAK-17585?focusedCommentId=16002705&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-16002705
         -->
-        <microprofile-metrics-api.version>2.3</microprofile-metrics-api.version>
+        <microprofile-metrics-api.version>3.0.1</microprofile-metrics-api.version>
         <testcontainers.version>1.16.3</testcontainers.version>
 
         <!-- Maven Plugins -->

--- a/quarkus/pom.xml
+++ b/quarkus/pom.xml
@@ -36,7 +36,7 @@
             Their names should match the name used by Quarkus BOM and the versions are automatically set when running the 'set-quarkus-version.sh' script.
         -->
         <resteasy.version>4.7.7.Final</resteasy.version>
-        <hibernate-orm.version>5.6.12.Final</hibernate-orm.version>
+        <hibernate-orm.version>5.6.14.Final</hibernate-orm.version>
         <mysql-jdbc.version>8.0.30</mysql-jdbc.version>
         <postgresql-jdbc.version>42.5.0</postgresql-jdbc.version>
         <wildfly-common.version>1.5.4.Final-format-001</wildfly-common.version>

